src/backend/utils/storage.js

```diff
@@ -5,21 +5,52 @@ const FEED_ID = 'feed_id';
 const FEEDS = 'feeds';
 
 const POSTS = 'posts';
+const INACTIVE_FEEDS = 'inactive_feeds';
 
 module.exports = {
-  addFeed: async (name, url) => {
-    // "If the key does not exist, it is set to 0 before performing the operation"
-    // https://redis.io/commands/INCR
-    const feedId = await redis.incr(FEED_ID);
-    await redis
-      .multi()
-      // Using hmset() until hset() fully supports multiple fields:
-      // https://github.com/stipsan/ioredis-mock/issues/345
-      // https://github.com/luin/ioredis/issues/551
-      .hmset(feedId, 'name', name, 'url', url)
-      .sadd(FEEDS, feedId)
-      .exec();
-    return feedId;
+  /**
+   * Adds/Moves feed to the active FEEDS set and returns the feedId
+   * @param feed Feed Object
+   * @return feedId
+   */
+  addFeed: async feed => {
+    if (feed.feedId && (await redis.exists(INACTIVE_FEEDS)) === 1) {
```

Comment:
These `add*` functions are doing two things, and should be split.  First, you're dealing with creating a hash to store the feed data.  That makes sense, and should stay here.  Then, you're putting the id into a set, and that can get moved out to separate functions.

Also, you probably want to use [sismember](https://redis.io/commands/sismember) vs. exists.

Reply:
You're right, I think I was trying to do too much with this function. the exists is checking whether the set exists in the first place to see if we need to do SADD operation first.

Would `addFeed` even be an appropriate name for this function, since we're not technically adding the feed anywhere?